components/infra-proxy-service/server/v1/cookbooks.go

```diff
@@ -182,3 +219,57 @@ func fromAPIToListAvailableCookbooks(al chef.CookbookListResult) []*response.Coo
 
 	return cl
 }
+
+func fromSearchAPIToCookbookNodes(sr chef.SearchResult) []*response.NodeAttribute {
+	results := make([]*response.NodeAttribute, len(sr.Rows))
+	index := 0
+	for _, element := range sr.Rows {
+		node := getNodeAttributeFromRes(element)
+		results[index] = &response.NodeAttribute{
+			Name:        node.Name,
+			CheckIn:     node.CheckIn,
+			ChefGuid:    node.ChefGUID,
+			Environment: node.ChefEnvironment,
+			Platform:    node.Platform,
+			PolicyGroup: node.PolicyGroup,
+			Uptime:      node.Uptime,
+		}
+		index++
+	}
+
+	return results
+}
+
+func getNodeAttributeFromRes(data interface{}) NodeAttribute {
+	m := data.(map[string]interface{})["data"].(map[string]interface{})
+	node := NodeAttribute{}
+	if name, ok := m["name"].(string); ok {
+		node.Name = name
+	}
+
+	if checkin, ok := m["ohai_time"].(string); ok {
```

Comment:
At present `ohai_time` is not being populated in response. Looking for help?

Also, any code optimization are super welcome :)

Reply:
we discussed this a bit in slack - superteam channel:
```
lancewf: 
The "ohai_time" is stored in the node-attribute index.
We don't place it in the run, but we could without much work.
You can see the ohai_time on the "Attributes" tab on the node runs view if you search for it.  

dan 
I'm not sure we need it if we have some other timestamp.
```

Reply:
could you speak to why you were looking to use ohai_time instead of just the check in time @vsingh-msys ?

Reply:
Sorry, I forgot he was pulling this directly from Chef Server. The "ohai_time" is not at the root of the data element it is at "data" -> "automatic" -> "ohai_time". Here is an example. 
https://github.com/chef/automate/blob/82772a8c0a82da983bb9b329fb31716447d66609/components/ingest-service/examples/converge-success-report.json#L2684

Reply:
In the cookbook affected node list check-in time is one of the attributes that require to show.

In client-runs nodes, we consider `check-in` time as last run/end time and it sends over the data collector.

https://github.com/chef/chef/blob/633c4445e00f3174d156ac69a05be117a3fd0456/lib/chef/data_collector/run_end_message.rb#L63-L64

https://github.com/chef/automate/blob/0de8065086367dadc8e99b42d2b9b54b5b1d1657/components/ingest-service/backend/chef_client_run.go#L117-L119



In Chef Server we have only `ohai_time` which is close to check-in time. Also in chef manage, we showed ohai_time as a check-in time.

Ref: 
![node-check-in](https://user-images.githubusercontent.com/44496937/75841505-e940de00-5df3-11ea-9e3a-7b2ba0b9ffab.png)

Reply:
Here the problem is ohai_time is float and I am not able to propolate in gRPC API or golang parsing side.

In the demo, script https://gist.github.com/vsingh-msys/f53921a2d4eb89e3a41ad772bff253a0 is getting populated.

Reply:
This could be one of the problems, the demo is using "github.com/go-chef/chef" and the automate code is using "github.com/chef/go-chef".

Reply:
okk i see. i can't fully run the example code b/c there's a key needed, but if the problem is getting the float to a string, you should be able to use strconv for that:
```
	ohai := 1469581742.9826262
	fmt.Println(reflect.TypeOf(ohai))

	stringOhai := strconv.FormatFloat(ohai, 'E', -1, 64)
	fmt.Println(reflect.TypeOf(stringOhai))
```

Reply:
in any case, i'm putting my +1 on the pr. feel free to merge as is and make a card to come back to this!

Reply:
@vjeffrey thankyou so much!!! it is working as expected.

<img src="https://media.giphy.com/media/WvureNOQfLwmQ/giphy.gif" width="200" height="200" />